vscode-dotnet-sdk-extension/src/extension.ts

```diff
@@ -0,0 +1,167 @@
+
+    const displayWorker = extensionContext ? extensionContext.displayWorker : new WindowDisplayWorker();
+    const extensionConfigWorker = new ExtensionConfigurationWorker(extensionConfiguration, undefined);
+    const issueContext = (errorConfiguration: ErrorConfiguration | undefined, commandName: string, version?: string) => {
+        return {
+            logger: loggingObserver,
+            errorConfiguration: errorConfiguration || AcquireErrorConfiguration.DisplayAllErrorPopups,
+            displayWorker,
+            extensionConfigWorker,
+            eventStream,
+            commandName,
+            version,
+        } as IIssueContext;
+    };
+    const timeoutValue = extensionConfiguration.get<number>(configKeys.installTimeoutValue);
+    let storagePath: string;
+    if (os.platform() === 'win32') {
+        // Install to %AppData% on windows to avoid running into long path errors
+        storagePath = process.env.APPDATA ? process.env.APPDATA : context.globalStoragePath;
+    } else {
+        storagePath = context.globalStoragePath;
+        if (!fs.existsSync(storagePath)) {
+            fs.mkdirSync(storagePath);
+        }
+    }
+
+    const acquisitionWorker = new DotnetCoreAcquisitionWorker({
+        storagePath,
+        extensionState: context.globalState,
+        eventStream,
+        acquisitionInvoker: new AcquisitionInvoker(context.globalState, eventStream),
+        installationValidator: new InstallationValidator(eventStream),
+        timeoutValue: timeoutValue === undefined ? defaultTimeoutValue : timeoutValue,
+    });
+    const versionResolver = new VersionResolver(context.globalState, eventStream);
+
+    const dotnetAcquireRegistration = vscode.commands.registerCommand(`${commandPrefix}.${commandKeys.acquire}`, async (commandContext: IDotnetAcquireContext) => {
+        const pathResult = callWithErrorHandling(async () => {
+            eventStream.post(new DotnetSDKAcquisitionStarted());
+
+            let version: string | undefined = commandContext ? commandContext.version : undefined;
+            if (!version) {
+                version = await vscode.window.showInputBox({
+                    placeHolder: '5.0',
+                    value: '5.0',
+                    prompt: '.NET version, i.e. 5.0',
+                });
+            }
+            if (!version) {
+                displayWorker.showErrorMessage('No .NET SDK version provided', () => { /* No callback needed */ });
+                return undefined;
+            }
+
+            eventStream.post(new DotnetAcquisitionRequested(version!));
+            const resolvedVersion = await versionResolver.getFullSDKVersion(version!);
+            const dotnetPath = await acquisitionWorker.acquireSDK(resolvedVersion);
+            displayWorker.showInformationMessage(`.NET SDK ${version} installed to ${dotnetPath.dotnetPath}`, () => { /* No callback needed */ });
+            setPathEnvVar(path.dirname(dotnetPath.dotnetPath));
```

Comment:
Pull the `path.dirname(dotnetPath.dotnetPath)` into its own variable to ease debugability